src/Microsoft.AspNetCore.Rewrite/UrlRewriteBuilder.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            options.Rules.Add(new PathRule { MatchPattern = new Regex(regex), OnMatch = newPath, OnCompletion = stopRewriteOnSuccess ? Transformation.TerminatingRewrite : Transformation.Rewrite });
+            return options;
+        }
+
+        public static UrlRewriteOptions RewriteScheme(this UrlRewriteOptions options, bool stopRewriteOnSuccess = false)
+        {
+            options.Rules.Add(new SchemeRule {OnCompletion = stopRewriteOnSuccess ? Transformation.TerminatingRewrite : Transformation.Rewrite });
+            return options;
+        }
+
+        public static UrlRewriteOptions RedirectPath(this UrlRewriteOptions options, string regex, string newPath, bool stopRewriteOnSuccess = false)
+        {
+            options.Rules.Add(new PathRule { MatchPattern = new Regex(regex), OnMatch = newPath, OnCompletion = Transformation.Redirect });
+            return options;
+        }
+
+        public static UrlRewriteOptions RedirectScheme(this UrlRewriteOptions options, int? sslPort)
+        {
+            options.Rules.Add(new SchemeRule { SSLPort = sslPort, OnCompletion = Transformation.Redirect });
+            return options;
+        }
+
+        public static UrlRewriteOptions CustomRule(this UrlRewriteOptions options, Func<UrlRewriteContext, RuleResult> onApplyRule, Transformation transform, string description = null)
+        {
+            options.Rules.Add(new FunctionalRule { OnApplyRule = onApplyRule, OnCompletion = transform});
+            return options;
+        }
+
+        public static UrlRewriteOptions RulesFromConfig(this UrlRewriteOptions options, IConfiguration rulesFromConfig)
+        {
+            // TODO figure out naming
+            var rules = rulesFromConfig.GetSection("Rewrite").GetChildren();
+            // TODO eventually delegate this to another method.
+            foreach (var item in rules)
+            {
+                switch (item["type"])
+                {
+                    case "RewritePath":
+                        RewritePath(options, item["match"], item["action"], item["terminate"] == "true");
+                        break;
+                    case "RewriteScheme":
+                        RewriteScheme(options, item["terminate"] == "true");
+                        break;
+                    case "RedirectPath":
+                        RedirectPath(options, item["match"], item["action"], item["terminate"] == "true");
+                        break;
+                    case "RedirectScheme":
+                        int i; // TODO is this the right style here?
+                        RedirectScheme(options, Int32.TryParse(item["port"], out i) ? i : (int?)null);
+                        break;
+                    default:
+                        break;
+                }
+            }
+            return options;
+        }
+
+        public static UrlRewriteOptions AddFileProvider(this UrlRewriteOptions options, IFileProvider fileProvider)
```

Comment:
Remove this method. There shouldn't be anything called AddFileProvider. It's already a property on options that can be set.